feat: ✨ Add delete account functionality and error checking

zaverbac opened the pull request.

zaverbac committed:
feat(client): ✨ Add redirect, delete account button

Add ability for Catniss to delete their account and be redirected to splash page after acount deletion

zaverbac committed:
Merge branch 'development' of https://github.com/cat-acronym-team/cat-me-if-you-can into zaverbac/View-Account

zaverbac committed:
feat(client): ✨ Add Change password and Delete account button

Add two buttons and added css grid layout

zaverbac committed:
Merge branch 'development' of https://github.com/cat-acronym-team/cat-me-if-you-can into zaverbac/View-Account

zaverbac committed:
Merge branch 'zaverbac/deleteAccount' of https://github.com/cat-acronym-team/cat-me-if-you-can into zaverbac/deleteAccount

zaverbac committed:
feat(client): ✨ Add SignOut functionality and change password button

Add signout functionality and css along with change password button

zaverbac committed:
feat(client): ✨ Add link password to provider

Add ability for user to link password with their account and account for errors

zaverbac committed:
Merge branch 'development' of https://github.com/cat-acronym-team/cat-me-if-you-can into zaverbac/View-Account

zaverbac committed:
build: 📦️ Package-lock.json changed

package-lock.json had been change

zaverbac committed:
Merge branch 'development' of https://github.com/cat-acronym-team/cat-me-if-you-can into zaverbac/deleteAccount

zaverbac committed:
Merge branch 'zaverbac/View-Account' of https://github.com/cat-acronym-team/cat-me-if-you-can into zaverbac/deleteAccount

zaverbac committed:
feat(client): ✨ Add error checking

Add error checking to check if user signed in too long ago, if so, makes them sign in again

zaverbac committed:
fix(client): 🐛 Fix prettier bug

Fix prettier bug of missing space in comment of vite.config.ts

nstringham commented:
please change the PR title

nstringham commented on client/src/lib/firebase/auth.ts:
```diff
@@ -34,6 +37,30 @@ export async function loginAnonymous() {
+export function deleteAccount() {
+  const user = auth.currentUser;
+
+  if (user !== null) {
```
we should probably throw an error in the else case

nstringham commented on client/vite.config.ts.timestamp-1667075606205.mjs:
```diff
@@ -0,0 +1,8 @@
+// vite.config.ts
```
don't need this file

zaverbac replied:
Removed vite.config.ts.timestamp-1667075606205.mjs

nstringham replied:
Are you sure?

nstringham commented on functions/package-lock.json:
```diff
@@ -62,171 +62,87 @@
         "url": "https://opencollective.com/eslint"
```
lets not change this file like this

zaverbac replied:
Do you want me to change it back to anything? I dont know what was there before

nstringham replied:
Just revert it https://git-scm.com/docs/git-revert

zaverbac committed:
refactor(client): 🔥 Remove unrelated functionality

Remove unrelated functionality from deleteAccount branch

zaverbac committed:
feat(functions): ✨ Add deletion of user data

Add deletion of corresponding user document when user request account deletion

nstringham reviewed (changes requested):
oops forgot to submit my review

nstringham commented on functions/src/users.ts:
```diff
@@ -0,0 +1,7 @@
+import * as functions from "firebase-functions";
+import { userCollection } from "./firestore-collections";
+
+export const user = functions.auth.user().onDelete((user) => {
```
lets call this something that makes it more clear what it is doing like `onDeleteUser` or something like that

zaverbac replied:
got it

nstringham replied:
Why was this marked as resolved?

zaverbac replied:
Sorry, I made changes but forgot to push them

zaverbac committed:
Merge branch 'development' of https://github.com/cat-acronym-team/cat-me-if-you-can into zaverbac/deleteAccount

zaverbac committed:
revert(functions) "build: 📦️ Package-lock.json changed"

This reverts commit bbdff0f8cb13d70925fa2e7a6edc6c2b7fc9190e due to not wanting package-lock.json to be changed.

zaverbac committed:
build(client): 🔥 Remove vite.config.ts.timestamp file

Remove vite.config.timestamp file as it was unneeded

zaverbac committed:
refactor(functions): ♻️ Change function variable name

Change functionvariable name to onDeleteUser

zaverbac committed:
Merge branch 'zaverbac/deleteAccount' of https://github.com/cat-acronym-team/cat-me-if-you-can into zaverbac/deleteAccount

zaverbac committed:
refactor(client): 🔥 Remove linkUser credentials function

Remove linkUserCredentials function as it was unused

nstringham commented on client/src/routes/settings/+page.svelte:
```diff
@@ -1 +1,97 @@
+          <Actions>
+            <Button
+              defaultAction
+              use={[InitialFocus]}
```
don't need this because there is only one button

nstringham commented on client/src/routes/settings/+page.svelte:
```diff
@@ -1 +1,97 @@
+      >
+        <Header>
+          <Title id="confirm-account-deletion-title">Delete Account</Title>
+        </Header>
```
I think you can just use the title without the header

nstringham commented on client/src/routes/settings/+page.svelte:
```diff
@@ -1 +1,97 @@
+    height: auto;
+    width: 100%;
+  }
+</style>
```
this CSS doesn't do much

nstringham commented on client/src/routes/settings/+page.svelte:
```diff
@@ -1 +1,97 @@
+<html lang="en">
+  <h2>Account Settings</h2>
+  <main>
+    <div class="wrapper">
```
I don't think we really need this wrapper

nstringham commented on client/src/routes/settings/+page.svelte:
```diff
@@ -1 +1,97 @@
+        <Content id="deletion-confirmation-content">Are you sure?</Content>
+        <Actions>
+          <Button id="dont-delete-account">
+            <Label>No</Label>
```
maybe name the buttons "Delete Account" and "Cancle" so it is very clear what we are asking

nstringham commented on client/src/routes/settings/+page.svelte:
```diff
@@ -1 +1,97 @@
+            <Label>No</Label>
+          </Button>
+          <Button
+            id="confirm-delete-account"
```
Is this id used anywhere?

zaverbac committed:
refactor(client): 🔥 Remove unused css

Remove unused css grid styles as they were not needed

nstringham commented on client/src/routes/settings/+page.svelte:
```diff
@@ -1 +1,97 @@
+        </Header>
+        <Content id="deletion-confirmation-content">Are you sure?</Content>
+        <Actions>
+          <Button id="dont-delete-account">
```
is this id used anywhere?

nstringham commented on client/src/routes/settings/+page.svelte:
```diff
@@ -1 +1,84 @@
+          <Actions>
+            <Button
+              defaultAction
+              use={[InitialFocus]}
```
don't need these because there is only one button

nstringham commented on client/src/routes/settings/+page.svelte:
```diff
@@ -1 +1,84 @@
+              use={[InitialFocus]}
+              on:click={() => {
+                logOut();
+                goto("/");
```
log out already includes a goto

nstringham replied:
never mind

zaverbac committed:
refactor(client): ♻️ Change action button labels

Change action button labels to more meaningful names

zaverbac committed:
refactor(client): ♻️ Remove unused css ids

Remove unused css id as they werent being used. Also moved non-action buttons out of action

zaverbac committed:
refactor(client): ♻️ Remove wrapper

Remove wrapper div as it was not needed

nstringham commented on client/src/routes/settings/+page.svelte:
```diff
@@ -1 +1,75 @@
+    >
+      <Title id="confirm-account-deletion-title">Delete Account</Title>
+      <Content id="deletion-confirmation-content">Are you sure?</Content>
+      <Actions>
```
this id is unused

zaverbac committed:
refactor(client): ♻️ Remove unneccessary functionality, tags

Remove unused initialfocus function, style tag, and ID

nstringham commented:
related to #46
